libpromises/evalfunction.c

```diff
@@ -8513,10 +8513,27 @@ struct IsReadableThreadData
     FnCallResult result;
 };
 
+#ifndef HAVE_PTHREAD_CANCEL
+#define PTHREAD_CANCELED ((void *)-1)
+static void ThreadSignalHandler(int signum)
+{
+    pthread_exit(PTHREAD_CANCELED);
+}
+#endif
+
 static void *IsReadableThreadRoutine(void *data)
 {
     assert(data != NULL);
 
+#ifndef HAVE_PTHREAD_CANCEL
+    struct sigaction actions;
+    memset(&actions, 0, sizeof(actions));
+    sigemptyset(&actions.sa_mask);
+    actions.sa_flags = 0;
+    actions.sa_handler = ThreadSignalHandler;
+    sigaction(SIGUSR2, &actions, NULL);
```

Comment:
I'd prefer `SIGHUP` or some other signal that semantically matches the case? In fact, I think we should block many common signals in the new thread, we don't want handlers for `SIGINT` etc. to be handled in it. So I'd extend this to something like [what we do in cf-reactor](https://github.com/cfengine/nova/blob/master/cf-reactor/cf-reactor.h#L66-L82) and used one of the other signals here.

Reply:
Sure, I agree with SIGHUP/etc.

The reason for splitting commits was to acknowledge and make clear the use of another persons contribution.

I do agree that it makes more sense to have commits which are sufficient and complete on their own but also recognize that sometimes we make "mistakes" in a commit and then have to come back and fix them, sort of like in this case here.